src/client/trackers/real-user-monitoring-for-performance.js

```diff
@@ -1,60 +1,67 @@
-		const lastEntry = entries[entries.length - 1];
-		largestContentfulPaint = lastEntry.renderTime || lastEntry.loadTime;
-	});
-	lcpPerformanceObserver.observe({type: 'largest-contentful-paint', buffered: true});
-
-	ttiPolyfill.getFirstConsistentlyInteractive().then(timeToInteractive => {
-
-		// Disconnect the observer once it no longer needs to observe the performance data
-		// @SEE: https://w3c.github.io/performance-timeline/#the-performanceobserver-interface
-		lcpPerformanceObserver.disconnect();
-
-		const navigation = performance.getEntriesByType('navigation')[0];
-		const { type, domInteractive, domComplete, responseStart, requestStart } = navigation;
-
-		// Proceed only if the page load event is a "navigate".
-		// @see: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/type
-		if (type !== 'navigate') return;
-
-		try {
-			const timeToFirstByte = responseStart - requestStart;
-			const firstPaint = performance.getEntriesByName('first-paint')[0].startTime;
-			const firstContentfulPaint = performance.getEntriesByName('first-contentful-paint')[0].startTime;
-			const context = {
-				firstPaint: Math.round(firstPaint),
-				firstContentfulPaint: Math.round(firstContentfulPaint),
-				timeToFirstByte: Math.round(timeToFirstByte),
-				domInteractive: Math.round(domInteractive),
-				domComplete: Math.round(domComplete),
-				largestContentfulPaint: Math.round(largestContentfulPaint),
-				timeToInteractive: Math.round(timeToInteractive),
-			};
-
-			console.log(context); // eslint-disable-line no-console
-			const data = {
-				action: 'performance',
-				category: 'page',
-				...context
-			};
-			broadcast('oTracking.event', data);
+	// Proceed only if the page load event is a "navigate".
+	// @see: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/type
+	if (type !== 'navigate') return;
+
+	const context = {
+		action: 'performance',
+		category: 'page',
+		domInteractive: Math.round(domInteractive),
+		domComplete: Math.round(domComplete),
+	};
+
+	/**
+	 * analyticsTracker()
+	 *
+	 * This function is called every time one of the performance events occurs.
+	 * The "final" event should be `firstInputDelay`, which is triggered by any "input" event (most likely to be a click.)
+	 * Once all the metrics are present, it fires a broadcast() to the Spoor API.
+	 */
+	let hasAlreadyBroadcast = false;
+	const analyticsTracker = (({ metricName, duration, data }) => {
+		if (hasAlreadyBroadcast) return;
+
+		if (duration) {
+			// Metrics with "duration":
+			// firstPaint, firstContentfulPaint, firstInputDelay and largestContentfulPaint
+			context[metricName] = Math.round(duration);
 		}
-		catch (error) {
-			console.error(error); // eslint-disable-line no-console
+		else if (metricName === 'navigationTiming') {
+			context.timeToFirstByte = Math.round(data.timeToFirstByte);
+		}
+
+		// Broadcast only if all the metrics are present
+		const contextContainsAllRequiredMetrics = [
```

Comment:
I'd recommend hoisting this list up to the top of the module so that it's clearly visible, a comment could link off to the spec too maybe?

Reply:
Good call, cheers.